src/coreclr/jit/gentree.cpp

```diff
@@ -16064,14 +16069,14 @@ bool GenTree::DefinesLocal(Compiler* comp, GenTreeLclVarCommon** pLclVarTree, bo
             }
         }
 
-        return destAddr->DefinesLocalAddr(comp, width, pLclVarTree, pIsEntire);
+        return destAddr->DefinesLocalAddr(comp, width, pLclVarTree, pIsEntire, pOffset);
     }
     // Otherwise...
     return false;
 }
 
-// Returns true if this GenTree defines a result which is based on the address of a local.
-bool GenTree::DefinesLocalAddr(Compiler* comp, unsigned width, GenTreeLclVarCommon** pLclVarTree, bool* pIsEntire)
+bool GenTree::DefinesLocalAddr(
+    Compiler* comp, unsigned width, GenTreeLclVarCommon** pLclVarTree, bool* pIsEntire, ssize_t* pOffset)
```

Comment:
If we're going to introduce `pOffset` here I guess it wouldn't be hard to remove `comp`, `width` and `pIsEntire` as parameters and do that checking after the function returns, from the (a) caller. I.e. add `pOffset` and LEA-handling to `IsLocalAddrExpr` and change this function to call it and check entirety.

I'm also ok with leaving it for a future change or even leaving it for `GT_ADDR`'s departure.

Reply:
> I'm also ok with leaving it for a future change

I would prefer that, the change just following this one will delete `IsLocalAddrExpr` and split `DefinesLocalAddr` into two overloads (the one computing the offset / the current one computing "entirety").